src/agent/src/sandbox.rs

```diff
@@ -403,6 +417,124 @@ impl Sandbox {
             }
         });
     }
+
+    #[instrument]
+    pub fn setup_shared_mounts(&self, c: &LinuxContainer, mounts: &Vec<SharedMount>) -> Result<()> {
+        let mut src_ctrs: HashMap<String, i32> = HashMap::new();
+        for shared_mount in mounts {
+            match src_ctrs.get(&shared_mount.src_ctr) {
+                None => {
+                    if let Some(c) = self.find_container_by_name(&shared_mount.src_ctr) {
+                        src_ctrs.insert(shared_mount.src_ctr.clone(), c.init_process_pid);
+                    }
+                }
+                Some(_) => {}
+            }
+        }
+
+        // If there are no shared mounts to be set up, return directly.
+        if src_ctrs.len() == 0 {
+            return Ok(());
+        }
+
+        let mounts = mounts.clone();
+        let init_mntns = fcntl::open(
+            "/proc/self/ns/mnt",
+            OFlag::O_RDONLY | OFlag::O_CLOEXEC,
+            Mode::empty(),
+        )?;
+        let dst_mntns_path = format!("/proc/{}/ns/mnt", c.init_process_pid);
+        let dst_mntns = fcntl::open(
+            dst_mntns_path.as_str(),
+            OFlag::O_RDONLY | OFlag::O_CLOEXEC,
+            Mode::empty(),
+        )?;
```

Comment:
Both init_mntns and dst_mntns are leaked here. You can use `fs::File::from_raw_fd` to wrap it up into a `File` so that it is closed properly.

Reply:
Sure, thanks! I didn't consider the issue of resource leaks, particularly file descriptors.